src/pspm_pp.m

```diff
@@ -96,6 +102,20 @@
     [sts, data.data, data.header.sr] = pspm_prepdata(data.data, filt);
     if sts == -1, return; end
     msg = sprintf('butterworth filter');
+  case 'leaky_integrator'
+    tau_sec = varargin{4}; 
+    if ~isnumeric(tau_sec)
+      warning('ID:invalid_input', 'Tau must be numeric.'); return;
+    end
+    % Convert tau from seconds to samples
+    sample_rate = data.header.sr;  % Assuming the sample rate is stored here
+    tau_samples = tau_sec * sample_rate;
```

Comment:
I think you need to `round` here and make sure the result is not zero. You could simply use `pspm_time2index`.